### ISSUE TITLE
Demo the "in-progress" and "failed" message state

### ISSUE THREAD

**Issue by j3k0** [j3k0/starling-chat-console#6]
To demo the "in-progress" and "failed" message state, dummy server could add new messages as "in-progress" and set them "success" (or sometimes "failed") after 2 seconds.

`in-progress` should show some kind of loading indicator (a rotating stuff)

`failed` should show a failure indicator (a red "x" for instance)

**Comment by j3k0** [comment]
@JustinWalkerThotkraft While reviewing the code, I found out a little misunderstanding about this:

`ChatMessage` have a `_state` field, which can be either `STATE_IN_PROGRESS`, `STATE_SUCCESS` or `STATE_FAILED`.

What was asked is that the progress indicator shows when `_state` equals to `STATE_IN_PROGRESS` (or when the avatar is being loaded, as you did already). The failure shows when `_state` equals to `STATE_FAILED`.

When the avatar can't be loaded (or is null), would it be possible to provide a replacement BitMap for each message?

**Comment by JustinWalkerThotkraft** [comment]
I can provide a replacement bitmap for each message.  I just need the resource uri. 
Im sorry Im confused I think that functionality is in place, the changing of the states to success or failure occurs when the event is heard from the AvatarImage, in the ChatMessage class.

**Comment by j3k0** [comment]
No, I'm sorry if I'm not clear. The states of success or failure should not reflect the state of the AvatarImage, but of a message being send from server.

Example flow:
1. User types "Hello" in the reply area.
2. Server notifies of a change in the list of data.
3. `Server.getData()`'s last message is "Hello" with state = `STATE_IN_PROGRESS`
   - which means the request has been sent to the remote server but server hasn't replied yet.
4. If the request succeeds:
   - Server notifies of a change in the list of data
   - `Server.getData()`'s last message is "Hello" with state = `STATE_SUCCESS`
5. If the requests fails:
   - last message is "Hello" with state = `STATE_FAILED`

The UI's in-progress and failure icon have to reflect those changes of state.

**Comment by JustinWalkerThotkraft** [comment]
I understand.  I will have a change soon.

**Comment by JustinWalkerThotkraft** [comment]
After some thought Ive come to the conclusion that this will require a change to the server interface.
The client should send and store a unique id with the chat message to check against the returned chat message object.  

If a received chat message has this id then the sending was successful.

I'm making the chagne now nd will submit upon confirmation of this change.

**Comment by JustinWalkerThotkraft** [comment]
The requested change is in, the load state is now denoted in the upper right corner of the image.

**Comment by j3k0** [comment]
Instead of a URI for the replacement bitmap, would be nice to be able to provide it as a Texture (or Image). In my use case.

**Comment by JustinWalkerThotkraft** [comment]
Im sorry, I think I misunderstood, do you wish for me to provide the graphic.  or an interface with which to supply a default graphic for failed images, or both.

**Comment by j3k0** [comment]
I meant an interface that allows to provide a default graphics.

This graphics can be used if there is no avatar image (and eventually also when the image is being loaded).

**Comment by JustinWalkerThotkraft** [comment]
Sorry for the delayed response, the default loading and loafFailed texture interfaces are in.
They are apart of the ChatMessageDisplayConfig class.  
In TestHarness, there is an AssetManager on line:34.
Here you can define your texture locations.
In TestHarness on Line: 50, you can redefine your texture names.
